### DISCUSSION TITLE
chi_square error

### DISCUSSION THREAD

**Discussion by Pedro-Floriano** [mboquien/cigale#24] [General]
Hi, this is my first time running cigale, i've followed all the manual instructions but I'm still getting the same error, even trying to run the sdss_qso example:
```
[ERROR] No suitable model found for [all my objects]. It may be that models are older than the universe or that the χ² are very 
large.
```

This is the data I'm working with: 
[sed_data.txt](https://github.com/mboquien/cigale/files/9532217/sed_data.txt)
This is the pcigale.ini:
[pcigale_ini.txt](https://github.com/mboquien/cigale/files/9532243/pcigale_ini.txt)

And this is the full output:
![Screenshot from 2022-09-09 01-00-26](https://user-images.githubusercontent.com/113158126/189269735-71c616e3-13de-4cba-bc41-772de9d57f51.png)

**Comment by Guang91** [comment]
Let me try to reproduce your error. Can you share your filter files as well? (those are not included in CIGALE by default)

**Reply by Pedro-Floriano** [reply]
Sure: 
[SPLUS_filters.zip](https://github.com/mboquien/cigale/files/9536746/SPLUS_filters.zip)
Thank you, in advance.

**Reply by Guang91** [reply]
The issue is related to the AGN and X-ray modules. In brief, for the parameters you give, CIGALE can't find a physical model for your sources. 

The easiest solution is to remove the X-ray module. I don't know why you need it, as you don't have any X-ray data.

**Reply by Pedro-Floriano** [reply]
Ok, I removed the x-ray and agn modules and now that message isn't showing anymore, but when I try running:
``
pcigale-plots sed
``
It gives:
``
[WARNING] No SED found for [object]. No plot created.
``
And so no plot is created, I'm testing with removing other modules, but the error persevered. Is there anything else that could be wrong in pcigale.ini or with the data I'm using?

**Reply by Pedro-Floriano** [reply]
I just noticed I had "save_best_sed" as "False" now everything is working fine. Thank you for your help.

**Reply by getrude386** [reply]
Hi, I get the same error even when have "save_best_sed" as "True":


![Screenshot 2022-09-13 at 10 13 24](https://user-images.githubusercontent.com/105718721/189848319-b66fd98f-f6c3-40df-b338-c605ba534f78.png)



The results.txt file contains "nans" only

**Reply by mboquien** [reply]
You are computing a single model. It is not unlikely that it is not adequate for your objects.

**Reply by getrude386** [reply]
What could be the reason for computing 1 model when I have two sources? I have attached my pcigale.ini file
[pcigale_ini.txt](https://github.com/mboquien/cigale/files/9557347/pcigale_ini.txt)

**Reply by mboquien** [reply]
From what I see, you only have 1 value for each parameter, so the Cartesian product yields a single model. You need to give different values to build a grid of models.

**Reply by getrude386** [reply]
I now have 27216 models with the same error, and the results.txt file contains nan values only:

![Screenshot 2022-09-14 at 09 43 16](https://user-images.githubusercontent.com/105718721/190092763-4c189b99-1839-4166-8cce-0daae49ce070.png)

**Reply by mboquien** [reply]
Have you set the option to save the best fit in the `pcigale.ini` file?

**Reply by getrude386** [reply]
Yes, I managed to get the plots, but with wrong results. Still trying to figure out what could be the reason, because with other fitting tools, the observed and fluxes do fit.
![Screenshot 2022-09-20 at 10 03 56](https://user-images.githubusercontent.com/105718721/191202961-c8bccce2-1c4f-4904-b9ba-b00c39de8597.png)

**Reply by Guang91** [reply]
What's your lim_flag in pcigale.ini? Please set it to none

**Reply by getrude386** [reply]
It was nonscaling, but now I have changed it to none, the results are the same

**Reply by Guang91** [reply]
Interesting. @mboquien, any idea about the cause?

**Reply by mboquien** [reply]
Can you send us the latest version of 1) input file, 2) `pcigale.ini`, 3) `pcigale.ini.spec`, and 4) any non-standard filter?

**Reply by getrude386** [reply]
I couldn't find the quickest way of transferring the files from ilifu to my computer, I was copying them.

[data.txt](https://github.com/mboquien/cigale/files/9607939/data.txt)
[pcigale.ini.txt](https://github.com/mboquien/cigale/files/9607940/pcigale.ini.txt)
[pcigale.ini.spec.txt](https://github.com/mboquien/cigale/files/9607941/pcigale.ini.spec.txt)

**Reply by mboquien** [reply]
Thanks. I have looked at your files and I am surprised it runs at all as there are multiple issues that should lead to a crash.

1. The list of module lacks redshifting, even though it is mandatory
2. The module configuration is nevertheless listed but the redshift is forced to 0
3. There are bands in _err in the bands parameter of the analysis module
4. The input file contains an extra column
5. The VLA and Herschel PACS fluxes appear very inconsistent with the rest of the data

I made some quick adjustments so that the code at least runs and removing the VLA and PACS fluxes leads to quite OK results.
![image](https://user-images.githubusercontent.com/444815/191280482-f6a5544f-7621-4a44-9862-13df4f120890.png)

I do not see any obvious problem with the code here, so I would suggest to rework the configuration files and double check the photometric catalog.

**Reply by getrude386** [reply]
Looking at the issues that you have mentioned (1 to 4), I think they came with the copying process, because when I check the original files they are ok. Let me check number 5 and see what I will get.

**Reply by getrude386** [reply]
Just removed the VLA and PACS and I got better results, it means the problem is with the photometric data. Thank you

What I don't understand is, under "redshifting", it says I must leave it empty for the redshifts in the input file to be used, when I leave it empty I don't get models/SEDs, I only get them when I set redshifting to 0.

**Reply by mboquien** [reply]
This is because your objects are at redshifts significantly larger than 0 whereas your stellar populations have an age of 13 Gyr (so, older than the age of the universe at that redshift, which is not possible). If you reduce the age to values smaller than the age of the universe, then it works as expected.

**Reply by getrude386** [reply]
Ok, I see now, Thank you for the assistance  and taking time to respond to my million messages, much appreciated.

**Comment by qian-yang** [comment]
Hi, I got the same error when I were testing the code using the example data and ini file (sdss_qso).
<img width="827" alt="Screen Shot 2022-09-19 at 7 53 51 PM" src="https://user-images.githubusercontent.com/5937805/191138926-e1bd1d15-b055-4f38-b322-3ab07f696ca7.png">

**Reply by mboquien** [reply]
Which version of CIGALE are you using?

@Guang91 I just tried the `sdss_qso` example and the manual does not seem completely up-to-date with respect to the `pcigale.ini` file. Would you mind having a look? Thanks!

**Reply by qian-yang** [reply]
> Which version of CIGALE are you using?
> 
> @Guang91 I just tried the `sdss_qso` example and the manual does not seem completely up-to-date with respect to the `pcigale.ini` file. Would you mind having a look? Thanks!

I am using the latest version 2022.1 (July 4th, 2022).

**Comment by Guang91** [comment]
Yes, the example in the manual is outdated. Let me update it later today or
tomorrow.

**Reply by qian-yang** [reply]
Many thanks to all of you! I appreciate your quick responses.

**Reply by Guang91** [reply]
I just updated the example files. Could you try them and see?

**Reply by qian-yang** [reply]
It works! Thank you so much!
